api/services/RichText.js

```diff
@@ -1,54 +1,152 @@
+    if (el.textContent.length > MAX_LINK_LENGTH) {
+      el.innerHTML = `${el.textContent.slice(0, MAX_LINK_LENGTH)}…`
+    }
+
+    // Normalize legacy Mentions
+    if (
+      el.getAttribute('data-entity-type') === 'mention'||
+      el.getAttribute('data-user-id')
+    ) {
+      const newSpanElement = dom.createElement('span')
+
+      newSpanElement.className = 'mention'
+      newSpanElement.setAttribute('data-type', 'mention')
+      newSpanElement.setAttribute('data-id', el.getAttribute('data-user-id'))
+      newSpanElement.setAttribute('data-label', el.textContent)
+      newSpanElement.innerHTML = el.innerHTML
+
+      el.parentNode.replaceChild(newSpanElement, el)
+
+      return
+    }
+
+    // Normalize legacy Topics
+    if (
+      el.getAttribute('data-entity-type') === '#mention' ||
+      (!el.getAttribute('href') && el.textContent[0] === '#')
+    ) {
+      const newSpanElement = dom.createElement('span')
 
-  return $.html()
+      newSpanElement.className = 'topic'
+      newSpanElement.setAttribute('data-type', 'topic')
+      newSpanElement.setAttribute('data-id', el.textContent?.slice(1))
+      newSpanElement.setAttribute('data-label', el.textContent)
+      newSpanElement.innerHTML = el.innerHTML
+
+      el.parentNode.replaceChild(newSpanElement, el)
+
+      return
+    }
+  }, dom.querySelectorAll('a'))
+
+  return dom.querySelector('body').innerHTML
 }
 
 /*
-Returns a set of unique IDs for any mention members
-found in the provided HTML
 
-Used for generating notifications
+Prepares content for HTML Email delivery
+
+- Always make sure `processHTML` was ran first, this is done
+  in `Post#details()` and `Comment#text()`
+
+- This same logic is handled dynamcially on Web in `ClickCatcher`,
+  and on Mobile in the `HyloHTML` component.
+
 */
-export const getUserMentions = html => {
-  if (!html) return []
+export function qualifyLinks (processedHTML) {
+  const dom = getDOM(processedHTML)
+
+  // Convert Mention and Topic `span` elements to `a` elements
+  const convertSpansToAnchors = forEach(el => {
+    const anchorElement = dom.createElement('a')
+    let href = el.className === 'mention'
+      ? PathHelpers.mentionPath(el.getAttribute('data-id'), groupSlug)
+      : PathHelpers.topicPath(el.getAttribute('data-id'), groupSlug)
+
+    for (const attr of el.attributes) {
+      anchorElement.setAttribute(attr.name, attr.value)
+    }
 
-  let $ = Cheerio.load(html)
+    anchorElement.innerHTML = el.innerHTML
+    anchorElement.setAttribute('href', href)
 
-  return _.uniq($('a[data-user-id]').map(function () {
-    return $(this).attr('data-user-id').toString()
-  }).get())
+    el.parentNode.replaceChild(anchorElement, el)
+  })
+  convertSpansToAnchors(dom.querySelectorAll(
+    'span.topic, span.mention'
+  ))
+
+  forEach(el => {
+    const href = el.getAttribute('href')
+    if (href && !href.match(/^(https?|email):/)) {
+      el.setAttribute('href', Frontend.Route.prefix + href)
```

Comment:
Re. `FrontEnd.Route` It would be really nice to merge all/most of Route/Path helper things into `hylo-shared` for keeping Web, Mobile, and backend stuff in sync and coherence.